### PR TITLE
Add UIForeground.

### PR DESCRIPTION
**Pull request by SAnatoliiS**

### CODE REVIEW

**Comment by SAnatoliiS** [comment]
image

**Comment by savelichalex** [comment]
Is it just me, or `foreground` associates to a data fetching? :D

File: casts/rows/src/index.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 export * from './UIAccountRow';
 export * from './UILink';
 export * from './UICurrencyRow';
+export * from './UIForeground';
```
Comment (savelichalex) [inline casts/rows/src/index.ts]: It shouldn't be in `rows`. `rows` package is for `rows` that is used in `VirtualizedList` and it's ancestors. Such component only will provide confusion when to use the package.

File: casts/rows/src/UIForeground/Parts/SecondaryPart.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        return {
+            disabled,
+            negative,
+            partType: 'Secondary',
+            partState: onPress || disabled !== undefined ? 'Pressable' : 'NonPressable',
+        };
+    }, [disabled, negative, onPress]);
+
+    return (
+        <TouchableOpacity
```
Comment (savelichalex) [inline casts/rows/src/UIForeground/Parts/SecondaryPart.tsx]: So it have unnecessary `TouchableOpacity` (with all it's complications) even if `onPress` won't be ever provided?

File: casts/rows/src/UIForeground/UIForeground.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    Container,
+
+    PrimaryPart,
+    SecondaryPart,
+
+    ActionElement,
+    IconElement,
+    NumberElement,
+    SectionElement,
+    TextElement,
```
Comment (savelichalex) [inline casts/rows/src/UIForeground/UIForeground.tsx]: Too verbose if you ask me, you can omit `Element` without loosing meaning (btw it contradicts with React terminology, where `element` means a component presented in JSX, aka the one that `React.createElement` returns)

File: casts/rows/src/UIForeground/UIForeground.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+export function SectionElement() {
+    return <View />;
+}
+
+export const UIForeground: UIForegroundType = {
+    Container,
+
+    PrimaryPart,
+    SecondaryPart,
```
Comment (savelichalex) [inline casts/rows/src/UIForeground/UIForeground.tsx]: IMO `Part` is also unnecessary here

**Comment by SAnatoliiS** [comment]
This is an internal component that will be used in many popups

File: casts/rows/src/index.ts (hunk 1 above)
Comment (SAnatoliiS) [reply casts/rows/src/index.ts]: Moved to kit.popups

File: casts/rows/src/UIForeground/UIForeground.tsx (hunk 3 above)
Comment (SAnatoliiS) [reply casts/rows/src/UIForeground/UIForeground.tsx]: `Element` changed to `Cell`

File: casts/rows/src/UIForeground/UIForeground.tsx (hunk 4 above)
Comment (SAnatoliiS) [reply casts/rows/src/UIForeground/UIForeground.tsx]: `Part` changed to `Column`

File: kit/popups/src/UIForeground/Container.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    return <View style={styles.container}>{children}</View>;
+}
+
+const styles = StyleSheet.create({
+    container: {
+        /**
+         * The negative right margin is used to set the right padding of all child elements
+         * and not to think about the presence of an element on the right side.
+         */
+        marginHorizontal: -UILayoutConstant.contentInsetVerticalX3 / 2,
```
Comment (savelichalex) [inline kit/popups/src/UIForeground/Container.tsx]: I'm curious if a pressable area is correct on all platforms in that case

Comment (SAnatoliiS) [reply kit/popups/src/UIForeground/Container.tsx]: I checked it in all platforms

**Review by savelichalex — APPROVED** [review]

**Review by sertony — APPROVED** [review]

**Pull request merged**